cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1654,19 +1672,131 @@ String zkConnectionString(int connectToReplicas)  {
+                    acPromise.complete(ac);
+                } catch (RuntimeException e) {
+                    log.warn("Failed to create Admin Client. {}", e);
+                    acPromise.complete(null);
+                }
+                return acPromise.future();
+            });
+        }
+
+        /**
+         * If the kafka pod is running, change its configuration dynamically if it is possible (and needed).
+         * If it is not possible set the flag for specific podID an it will be rolled in {@code kafkaRollingUpdate}.
+         * @return Succeeded future
+         */
+        Future<ReconciliationState> kafkaBrokerDynamicConfiguration() {
+            // patches scale up/down situations
+            int runningReplicas = Math.min(oldKafkaReplicas, kafkaCluster.getReplicas());
+            List<Future> configFutures = new ArrayList<>(runningReplicas);
+
+            for (int podId = 0; podId < runningReplicas; podId++) {
+                int finalPodId = podId;
+                configFutures.add(podOperations.readiness(namespace, KafkaResources.kafkaPodName(name, podId), 1_000, operationTimeoutMs)
+                    .compose(ignore ->
+                            adminClient(adminClientProvider, namespace, name, finalPodId)
+                    .compose(ac -> {
+                        if (ac == null) {
+                            kafkaPodsUpdatedDynamically.put(finalPodId, null);
+                            log.warn("{} Could not create admin client.", reconciliation);
+                            return Future.succeededFuture();
+                        } else {
+                            log.debug("{} Determining kafka broker {} dynamic update ability", reconciliation, finalPodId);
+                            Promise<Void> p = Promise.promise();
+                            getCurrentConfig(finalPodId, ac)
+                                    .compose(res -> {
+                                        log.trace("Broker description {}", res);
+                                        KafkaBrokerConfigurationDiff configurationDiff = new KafkaBrokerConfigurationDiff(res, this.kafkaCm, kafkaCluster.getKafkaVersion(), finalPodId);
+                                        if (!configurationDiff.isEmpty()) {
+                                            try {
+                                                AlterConfigsResult alterConfigResult = ac.incrementalAlterConfigs(configurationDiff.getUpdatedConfig());
+                                                KafkaFuture<Void> brokerConfigFuture = alterConfigResult.values().get(new ConfigResource(ConfigResource.Type.BROKER, Integer.toString(finalPodId)));
+                                                return kafkaFutureToVertxFuture(brokerConfigFuture).compose(result -> {
+                                                    log.debug("{} Dynamic AlterConfig result for broker {} {}", reconciliation, finalPodId, result);
```

Comment:
Question: What does the result contain? The whole configuration? Some Success / failure statement?

Reply:
`get()` returns `KafkaFuture<Void>` so I think it will be always null.

Reply:
So I guess it should be updated?